terraform/main.tf

```diff
@@ -102,3 +102,297 @@ resource "google_storage_bucket_iam_binding" "binding" {
+        value_source {
+          secret_key_ref {
+            secret = "superset-database-password"
+            version = "1"
+          }
+        }
+      }
+      env {
+        name = "SUPERSET_SECRET_KEY"
+        value_source {
+          secret_key_ref {
+            secret = "superset-secret-key"
+            version = "1"
+          }
+        }
+      }
+      env {
+        name = "CLOUD_SQL_CONNECTION_NAME"
+        value_source {
+          secret_key_ref {
+            secret = "superset-database-connection-name"
+            version = "1"
+          }
+        }
+      }
+      env {
+        name = "AUTH0_CLIENT_ID"
+        value_source {
+          secret_key_ref {
+            secret = "superset-auth0-client-id"
+            version = "1"
+          }
+        }
+      }
+      env {
+        name = "AUTH0_CLIENT_SECRET"
+        value_source {
+          secret_key_ref {
+            secret = "superset-auth0-client-secret"
+            version = "2"
+          }
+        }
+      }
+      env {
+        name = "AUTH0_DOMAIN"
+        value_source {
+          secret_key_ref {
+            secret = "superset-auth0-domain"
+            version = "1"
+          }
+        }
+      }
+
+      ports {
+        container_port = 8088
+      }
+      resources {
+        limits = {
+          cpu    = "4"
+          memory = "2048Mi"
+        }
+      }
+    }
+    volumes {
+      name = "bucket"
+      gcs {
+        bucket    = google_storage_bucket.superset_storage.name
+        read_only = false
+      }
+    }
+    volumes {
+      name = "cloudsql"
+      cloud_sql_instance {
+        instances = ["catalyst-cooperative-pudl:us-central1:superset-database"]
+      }
+    }
+  }
+}
+
+resource "google_cloud_run_v2_service_iam_member" "noauth" {
+  location = google_cloud_run_v2_service.pudl-superset.location
+  name     = google_cloud_run_v2_service.pudl-superset.name
+  role     = "roles/run.invoker"
+  member   = "allUsers"
+}
+
+resource "google_secret_manager_secret" "superset_secret_key" {
+  secret_id = "superset-secret-key"
+  replication {
+    auto {}
+  }
+}
+
+resource "google_sql_database_instance" "postgres_pvp_instance_name" {
+  name             = "superset-database"
+  region           = "us-central1"
+  database_version = "POSTGRES_14"
+  settings {
+    tier = "db-custom-2-7680"
+    password_validation_policy {
+      min_length                  = 6
+      reuse_interval              = 2
+      complexity                  = "COMPLEXITY_DEFAULT"
+      disallow_username_substring = true
+      password_change_interval    = "30s"
+      enable_password_policy      = true
+    }
+  }
+  # set `deletion_protection` to true, will ensure that one cannot accidentally delete this instance by
+  # use of Terraform whereas `deletion_protection_enabled` flag protects this instance at the GCP level.
+  deletion_protection = true
+}
+
+resource "google_secret_manager_secret" "superset_database_username" {
```

Comment:
Hooray! I love that you're using the secret manager to manage secrets, it makes my shriveled little security heart sing.

It's a little heavyweight to define *all this boilerplate* just to say "there are these secrets, here are their names, and this service account has access to them." We can use the [`for_each`](https://developer.hashicorp.com/terraform/language/meta-arguments/for_each) syntax of Terraform to drastically reduce that.

I think we should do that *before* this gets merged, since it will make Terraform re-create all the secrets, which requires a little manual intervention to re-populate the secrets.

Reply:
Ah good to know! Do you think it's worth changing now?